### PR TITLE
Python tool to spawn one of each item / furniture / terrain / traps

### PR DESCRIPTION
**Pull request by Saicchi**
#### Summary
Infrastructure "Tool to spawn all items / furniture / terrain / traps"

#### Purpose of change
Allows tileset developers to view missing sprites.

#### Describe the solution
Add python tool to spawn those things, **starting at submap `0,0`**

#### Describe alternatives you've considered
Spawn them from in-game somehow.

#### Testing
Spawns everything, it looks chaotic but works properly.

#### Additional context

**This tool overwrites the map file on the world**, make sure it points to the right place. This also means that it won't clean up previous changes.

Usage:
`tileset_test.py folder maps_location`
`folder` -> where the tool will look for `.json` files
`maps_location` -> where the tool will save the maps it generates

`-f, --force` -> overwrite maps folder if it exists
`-n, --dry_run` -> print files to terminal

This was not tested on Windows, only Linux.

It also can't follow symbolic links, blame Python pathlib.

Some tiles might show up as just concrete, that is due to invisible furniture / traps / items, etc. Use the Debug Map Editor in the debug menu to view their ID.

Items

Furniture

Terrain

Traps : Things outside the concrete floors were spawned by the trap. If they're empty that is because they weren't revealed yet or the trap itself is empty, remember to check with the in-game debug map editor.

### CODE REVIEW

**Comment by actual-nh** [comment]
So one could limit it to items, furniture, or terrain by appropriate directory settings? That may be worth noting. (Thanks, BTW!)

**Comment by actual-nh** [comment]
Note: If I am reading the script correctly, you should add "traps" to the PR title and documentation.

**Comment by Saicchi** [comment]
Oh. I forgot it also added traps, thanks.

File: tools/tileset_test.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+class HandlerUnhandledType(Exception):
+    def __init__(self, key):
+        super().__init__(f"key not handled by Handler: {key}")
+
+
+class Handler:
+
+    def __init__(self, path: pathlib.Path, parsed_ids: dict):
+        self.maps_folder = pathlib.Path(args.path)
```
Comment (int-ua) [inline tools/tileset_test.py:285]: It looks quite good overall. Just a few minor improvements visible on the surface: args variable is not passed explicitly to the init here and the path parameter is not used.

**Comment by int-ua** [comment]
@Saicchi would you mind using single quotes everywhere? Right now quotes are mixed

**Comment by int-ua** [comment]
Also this PR could benefit from some screenshots ( include Ultica, please :) )

**Comment by Saicchi** [comment]
Updated it to fix the errors pointed out and use single quotes whenever possible. I finished the base version of this PR back in April and looking at it now, it looks very ugly... Kinda makes me want to rewrite it from scratch.

File: hibernate.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+# Steps to Reproduce
+
+# Expected behavior
+
+# Screenshots
+
+# Versions and configuration
+
+# Additional context
```
Comment (int-ua) [inline hibernate.md:13]:
```suggestion
```

Comment (Saicchi) [reply hibernate.md]: Oops. Thanks

**Comment by int-ua** [comment]
> I finished the base version of this PR back in April and looking at it now, it looks very ugly... Kinda makes me want to rewrite it from scratch.

It's a nice thing to strive for and it does not matter if the script gets merged already or later. Do you mean code style or architecture?

**Comment by Saicchi** [comment]
The coding style. I'd add more functions and classes to make things more organized since I got lost on the code when I got back to working on it. Though as it is adding a new feature to the file should be easy so I don't think it's that important since it already does well what is asked of it.

**Pull request merged**